engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java

```diff
@@ -1242,4 +1242,104 @@ public void shouldCreateCronTimer() {
+    final var deployedProcess =
+        engine
+            .deployment()
+            .withXmlResource(model)
+            .deploy()
+            .getValue()
+            .getProcessesMetadata()
+            .get(0);
+    final long processDefinitionKey = deployedProcess.getProcessDefinitionKey();
+
+    final TimerRecordValue timerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+            .withProcessDefinitionKey(deployedProcess.getProcessDefinitionKey())
+            .getFirst()
+            .getValue();
+    assertThat(timerRecord.getDueDate())
+        .isGreaterThan(start.plusSeconds(10).toInstant().toEpochMilli());
+
+    // when
+    final long nextDueDate = timerRecord.getDueDate() + 10_000L;
+    engine.increaseTime(Duration.ofSeconds(10));
+
+    // then
+    final TimerRecordValue lastTimerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+            .withProcessDefinitionKey(processDefinitionKey)
+            .skip(1)
+            .getFirst()
+            .getValue();
+    Assertions.assertThat(lastTimerRecord).hasDueDate(nextDueDate);
+
+    assertThat(
+            RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+                .withProcessDefinitionKey(processDefinitionKey)
+                .limit(2)
+                .count())
+        .isEqualTo(2);
+  }
+
+  @Test
+  public void shouldAvoidTriggeringMultipleTimes() {
+    // given
+    final ZonedDateTime start =
+        ZonedDateTime.ofInstant(Instant.now(), ZoneId.systemDefault()).plusMinutes(30);
+    final BpmnModelInstance model =
+        Bpmn.createExecutableProcess("process")
+            .startEvent("start")
+            .timerWithCycle(String.format("R3/%s/PT10M", start))
+            .endEvent("end")
+            .done();
+    final var deployedProcess =
+        engine
+            .deployment()
+            .withXmlResource(model)
+            .deploy()
+            .getValue()
+            .getProcessesMetadata()
+            .get(0);
+    final long processDefinitionKey = deployedProcess.getProcessDefinitionKey();
+
+    // when
+    engine.stop();
+    engine.increaseTime(Duration.ofHours(1));
+    RecordingExporter.reset();
+    engine.start();
+    final long now = engine.getClock().getCurrentTimeInMillis();
+
+    // then
+    final TimerRecordValue firstTimerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+            .withProcessDefinitionKey(processDefinitionKey)
+            .getFirst()
+            .getValue();
+    Assertions.assertThat(firstTimerRecord)
+        .hasDueDate(start.toInstant().toEpochMilli())
+        .hasTargetElementId("start")
+        .hasElementInstanceKey(TimerInstance.NO_ELEMENT_INSTANCE);
+
+    final TimerRecordValue secondTimerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+            .withProcessDefinitionKey(processDefinitionKey)
+            .skip(1)
+            .getFirst()
+            .getValue();
```

Comment:
❌ We don't want the second timer to trigger already. It should be scheduled 10 minutes into the future. And shouldn't trigger during this test case.

The reasoning is that during the downtime the timer should've triggered multiple times, but the engine was down. Then after the restart, it triggers immediately once and should schedule the next timer 10 minutes into the future. This would prevent the timer from triggering multiple times at startup.

You can test this by finding the second TimerIntent.CREATED timer record, and asserting that the due date lies in the future. Something like:
```java
final TimerRecordValue secondTimerRecord =
    RecordingExporter.timerRecords(TimerIntent.CREATED)
        .withProcessDefinitionKey(processDefinitionKey)
        .skip(1)
        .getFirst()
        .getValue();

final long now = engine.getClock().getCurrentTimeInMillis();
assertThat(secondTimerRecord.getDueDate()).isGreaterThan(now);
```

This will show that there is still something wrong with the current implementation because the timer is scheduled to trigger immediately, instead of being scheduled in the future.

Reply:
> immediately trigger a timer start event with cycle start that is in the past

---
>  We don't want the second timer to trigger already. It should be scheduled 10 minutes into the future. And shouldn't trigger during this test case.

❓ Hi @korthout. How do we distinguish between **user-defined past start time** and **past start times due to recovery after engine downtime**.

If it is not possible to distinguish the two behaviors, should we unify the behavior, when the user defines a start time in the past,  we will trigger after 10 minutes instead of immediately

Reply:
>when the user defines a start time in the past, we will trigger after 10 minutes instead of immediately

I don't think that would be the expected behavior from the user's perspective. 

>How do we distinguish between user-defined past start time and past start times due to recovery after engine downtime.

Do you mean in the implementation? Or in the test?

In the test, we can look at the timestamp of the timer records to see the actual time that the timer triggered or when it was created and look at the due date to see the time when it is supposed to be due (or was supposed to be due). These might differ, especially when the due date is passed while the engine is turned off.

In the implementation, I'm not sure yet. If you don't see a solution for it, please let me know.

Reply:
> Do you mean in the implementation? Or in the test?

Sorry, let me clarify another way, do we allow users to define a start time in the past (in C7 it is not allowed to define a start time less than the current time)

Reply:
Ah that's an interesting thought. It would reduce the scope 👍 And since the timer cycle with a start hasn't been released in a stable release yet (planned for 8.1), it's totally fine to take that approach.

Feel free to add a validation for that.

We can always see if we get any user-requests to allow timer cycles with a start in the past. It might be annoying to deploy older process models. Users would have to adjust them before they can be deployed. But we can improve this later.